packages/internal/bridge/bridge-sample-app/src/deposit.ts

```diff
@@ -38,22 +38,24 @@ async function deposit() {
   const rootBridge: ethers.Contract = getContract("RootERC20BridgeFlowRate", params.rootBridgeAddress, params.rootProvider);
   const childBridge: ethers.Contract = getContract("ChildERC20Bridge", params.childBridgeAddress, params.childProvider);
 
-  let rootBridgeChildAddress = await rootBridge.rootTokenToChildToken(params.rootToken);
-  let childBridgeChildAddress = await childBridge.rootTokenToChildToken(params.rootToken);
-
-  if (rootBridgeChildAddress === ethers.constants.AddressZero) {
-    console.log('token not mapped, please map token before depositing');
-    return;
-  }
-
-  if (childBridgeChildAddress === ethers.constants.AddressZero) {
-    console.log('token mapping incomplete, please wait for token to map to childBridge before depositing');
-    return;
-  }
-
-  if (rootBridgeChildAddress !== childBridgeChildAddress) {
-    console.log(`token mappings mismatch on rootBridge (${rootBridgeChildAddress}) & childBridge (${childBridgeChildAddress}).`, );
-    return;
+  if (params.childToken.toUpperCase() !== 'NATIVE' && params.rootToken.toUpperCase() !== 'NATIVE') {
+    let rootBridgeChildAddress = await rootBridge.rootTokenToChildToken(params.rootToken);
+    let childBridgeChildAddress = await childBridge.rootTokenToChildToken(params.rootToken);
+    
+    if (rootBridgeChildAddress === ethers.constants.AddressZero) {
+      console.log('token not mapped, please map token before depositing');
+      return;
+    }
+  
+    if (childBridgeChildAddress === ethers.constants.AddressZero) {
+      console.log('token mapping incomplete, please wait for token to map to childBridge before depositing');
+      return;
+    }
+  
+    if (rootBridgeChildAddress !== childBridgeChildAddress) {
+      console.log(`token mappings mismatch on rootBridge (${rootBridgeChildAddress}) & childBridge (${childBridgeChildAddress}).`, );
+      return;
+    }
```

Comment:
should we be return an error or throw if these conditions are true, given these are failure cases. Line 182 would also print "Exiting successfully" otherwise, which might be misleading?

Reply:
sure im throwing an error now here and in the same checks we do in the withdraw script